docs/dsl1.rst

```diff
@@ -0,0 +1,192 @@
+.. _dsl1-page:
+
+********************
+Migrating from DSL 1
+********************
+
+In Nextflow version ``22.03.0-edge``, DSL2 became the default DSL version. In version ``22.12.0-edge``,
+DSL1 support was removed, and this documentation was updated to use DSL2 by default. Users who are still
+using DSL1 should migrate their pipelines to DSL2 in order to use the latest versions of Nextflow. This page
+describes the differences between DSL1 and DSL2, and how to migrate to DSL2.
+
+In Nextflow versions prior to ``22.03.0-edge``, you must enable DSL2 explicitly in order to use it. You can either
+set the feature flag in your pipeline script::
+
+    nextflow.enable.dsl=2
+
+Or set the environment variable where you launch Nextflow::
+
+    export NXF_DEFAULT_DSL=2
+
+
+Processes and workflows
+=======================
+
+In DSL1, a process definition is also the process invocation. Process inputs and outputs are connected to channels
+using ``from`` and ``into``. Here is the :ref:`getstarted-first` example written in DSL1::
+
+    nextflow.enable.dsl=1
+
+    params.str = 'Hello world!'
+
+    process splitLetters {
+
+        output:
+        file 'chunk_*' into letters
+
+        """
+        printf '${params.str}' | split -b 6 - chunk_
+        """
+    }
+
+    process convertToUpper {
+
+        input:
+        file x from letters.flatten()
+
+        output:
+        stdout result
+
+        """
+        cat $x | tr '[a-z]' '[A-Z]'
+        """
+    }
+
+    result.view { it.trim() }
+
+To migrate this code to DSL2, you need to move all of your channel logic throughout the script
+into a (nameless) ``workflow`` definition. Additionally, you must call each process explicitly,
+passing any input channels as arguments (instead of ``from ...``) and receiving any output channels
+as return values (instead of ``into ...``).
+
+Refer to the :ref:`workflow-page` page to learn how to define a workflow, as well as the original
+:ref:`getstarted-first` example to see the resulting DSL2 version.
```

Comment:
Is it worth pasting it here as well? It'll be duplicated, but would be much easier to compare the two directly. Also it's likely that the getting-started example may slowly drift from this one.